test/extensions/filters/network/thrift_proxy/integration.cc

```diff
@@ -63,9 +63,26 @@ void BaseThriftIntegrationTest::preparePayloads(const PayloadOptions& options,
   };
 
   if (options.service_name_) {
+    args.push_back("-s");
     args.push_back(*options.service_name_);
   }
-  args.push_back("--");
+
+  if (options.headers_.size() > 0) {
+    args.push_back("-H");
+
+    std::stringstream ss;
+    size_t sz = options.headers_.size();
+    for (size_t i = 0; i < sz; i++) {
+      std::pair<std::string, std::string> p = options.headers_.at(i);
+      ss << p.first << "=" << p.second;
+      if (i < sz - 1) {
+        ss << ",";
+      }
+    }
```

Comment:
How about

```
std::vector<std::string> headers;
std::transform(options.headers_.begin(), options.headers_.end(), std::back_inserter(headers),
  [](std::pair<std::string, std::string>& header) -> std::string { return header.first + "=" + header.second; });
args.push_back(StringUtil::join(headers, ","));
```

Reply:
these are all things I didn't know existed so 👍

Reply:
Might need to `include <algorithm>` to get transform and back_inserter, FYI